pkg/operator/staticpod/startupmonitor/fallback.go

```diff
@@ -128,7 +136,52 @@ func (f *staticPodFallback) fallbackToPreviousRevision(reason, message string) e
 }
 
 func (f *staticPodFallback) markRevisionGood() error {
-	return f.createLastKnowGoodRevisionFor(f.revision, true)
+	if err := f.createLastKnowGoodRevisionFor(f.revision, true); err != nil {
+		return err
+	}
+
+	if f.operatorClient == nil || len(f.nodeName) == 0 {
```

Comment:
since both, kubeconfig and the node name are required we could get rid of this as well

Reply:
fixed